pipeline/07-export.R

```diff
@@ -450,33 +482,21 @@ assessment_card_prepped <- assessment_card %>%
   select(
     township_code, meta_pin, meta_card_num, meta_class, meta_nbhd_code,
     meta_card_pct_total_fmv, pred_card_initial_fmv, pred_card_final_fmv,
-    char_yrblt, char_beds, char_ext_wall, char_bsmt, char_bsmt_fin,
+    char_yrblt, char_beds, char_ext_wall, char_bsmt, char_bsmt_fin, char_air,
     char_heat, char_bldg_sf, char_type_resd, char_land_sf, char_apts, char_ncu
   ) %>%
   mutate(
     meta_pin = glue(
       '=HYPERLINK("https://www.cookcountyassessor.com/pin/{meta_pin}",
       "{meta_pin}")'
     ),
-    # Make sure the format of char_apts matches the short format we used to
-    # generate assessment_pin_prepped
-    char_apts = case_when(
-      char_apts == "NONE" | is.na(char_apts) ~ NA_character_,
-      char_apts == "TWO" ~ "2",
-      char_apts == "THREE" ~ "3",
-      char_apts == "FOUR" ~ "4",
-      char_apts == "FIVE" ~ "5",
-      char_apts == "FIX" ~ "6",
-      TRUE ~ "Missing"
-    ),
+    char_apts = format_char_apts(char_apts),
     # char_ncu should only apply to 212s
     char_ncu = ifelse(meta_class != "212", NA, char_ncu)
```

Comment:
I think this was an intentional change, but now I'm a little confused by it -- why are we filtering for 212s when processing `char_ncu`, but we're not filtering for 211s and 212s when processing `char_apts`? I've gone ahead and preserved that behavior but I'm not sure why.

Reply:
I think the filtering in the ingest stage now. All non-211/212 cards should have no `APTS` variable, same for `NCU` and 212s. This filtering is basically redundant.

Reply:
Huh, I think something is going on with this -- I'm noticing non-211/212 cards with `APTS` and `NCU` variables in the training and assessment sets. I'm also realizing that the new filtering I introduced doesn't fully cover this case, since the non-211/212 cards I'm noticing are all multi-card PINs where some of the chars have `APTS`/`NCU` and others don't, meaning that the aggregation and filter operations I introduced will preserve those values. I'm going to dig into this a little bit more tomorrow to see if I can figure out where it's coming from.